tokens/src/impls.rs

```diff
@@ -146,43 +162,48 @@ where
 	type Balance = B;
 
 	fn total_issuance() -> Self::Balance {
-		C::convert_balance(T::total_issuance(GetCurrencyId::get()), GetCurrencyId::get())
+		C::convert_balance_saturated(T::total_issuance(GetCurrencyId::get()), GetCurrencyId::get())
 	}
 
 	fn minimum_balance() -> Self::Balance {
-		C::convert_balance(T::minimum_balance(GetCurrencyId::get()), GetCurrencyId::get())
+		C::convert_balance_saturated(T::minimum_balance(GetCurrencyId::get()), GetCurrencyId::get())
 	}
 
 	fn balance(who: &AccountId) -> Self::Balance {
-		C::convert_balance(T::balance(GetCurrencyId::get(), who), GetCurrencyId::get())
+		C::convert_balance_saturated(T::balance(GetCurrencyId::get(), who), GetCurrencyId::get())
 	}
 
 	fn reducible_balance(who: &AccountId, keep_alive: bool) -> Self::Balance {
-		C::convert_balance(
+		C::convert_balance_saturated(
 			T::reducible_balance(GetCurrencyId::get(), who, keep_alive),
 			GetCurrencyId::get(),
 		)
 	}
 
 	fn can_deposit(who: &AccountId, amount: Self::Balance, mint: bool) -> DepositConsequence {
-		T::can_deposit(
-			GetCurrencyId::get(),
-			who,
-			C::convert_balance_back(amount, GetCurrencyId::get()),
-			mint,
-		)
+		let amount = C::convert_balance_back(amount, GetCurrencyId::get());
+		let amount = match amount {
+			Ok(amount) => amount,
+			Err(_) => return DepositConsequence::Overflow,
+		};
```

Comment:
Why is this different from tag1 below?

Reply:
there is `WithdrawConsequence::Underflow` but no `DepositConsequence::Underflow`
https://github.com/paritytech/substrate/blob/c0793b58c509f352b6006e402350ac9ad8017823/frame/support/src/traits/tokens/misc.rs#L93